packages/wallet-sdk/src/util/provider.ts

```diff
@@ -56,17 +56,19 @@ export function getCoinbaseInjectedProvider({
   metadata,
   preference,
 }: Readonly<ConstructorOptions>): ProviderInterface | undefined {
+  const { appName, appLogoUrl, appChainIds } = metadata;
+
   if (preference.options !== 'smartWalletOnly') {
     const extension = getCoinbaseInjectedLegacyProvider();
     if (extension) {
-      const { appName, appLogoUrl, appChainIds } = metadata;
-      extension.setAppInfo?.(appName, appLogoUrl, appChainIds);
+      extension.setAppInfo?.(appName, appLogoUrl, appChainIds, preference);
```

Comment:
adding it for extension too, just in case we need to use it, no harm even if we dont

Reply:
Did you test this on the extension side? Maybe include a video in the PR description?

Reply:
added demo!